src/main/kotlin/com/terraformation/backend/tracking/ObservationService.kt

```diff
@@ -486,6 +514,57 @@ class ObservationService(
     }
   }
 
+  /** Create monitoring plot, observation, and a claimed observation plot for the current user */
+  private fun createAdHocPlotObservation(
+      boundary: Polygon,
+      name: String,
+      observationType: ObservationType,
+      observedTime: Instant,
+      plantingSiteId: PlantingSiteId,
+  ): Pair<MonitoringPlotId, ObservationId> {
+    val userId = currentUser().userId
+    val now = clock.instant()
+    val observedDate = LocalDate.ofInstant(observedTime, ZoneOffset.UTC)
+
+    return dslContext.transactionResult { _ ->
+      val observationId =
+          observationStore.createObservation(
+              NewObservationModel(
+                  startDate = observedDate,
+                  endDate = observedDate,
+                  id = null,
+                  isAdHoc = true,
+                  observationType = observationType,
+                  plantingSiteId = plantingSiteId,
+                  state = ObservationState.InProgress,
+              ))
+
+      val monitoringPlotsRow =
+          MonitoringPlotsRow(
+              boundary = boundary,
+              createdBy = userId,
+              createdTime = now,
+              // Do we even need full name?
+              fullName = name,
+              isAvailable = false,
+              modifiedBy = userId,
+              modifiedTime = now,
+              name = name,
+              plantingSiteId = plantingSiteId,
+              sizeMeters = MONITORING_PLOT_SIZE_INT,
+          )
+
+      monitoringPlotsDao.insert(monitoringPlotsRow)
```

Comment:
I think we'll want to also insert a history row here (see my comment on the previous PR in the stack).

I'd probably put this in `PlantingSiteStore` to keep the code organization more consistent, since the store classes are where we usually put CRUD operations, especially ones that write data, and we're calling store classes for the other write operations in this function.